samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -426,57 +454,58 @@ void app_fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
 	}
 }
 
-void app_fp_adv_rpa_rotation_suspend(bool suspended)
+static void fp_adv_request_start(void)
 {
-	fp_adv_rpa_rotation_suspended = suspended;
+	bool has_account_key = bt_fast_pair_has_account_key();
+	bool requested_adv_on = fp_adv_trigger_is_enabled_any();
+	enum app_fp_adv_mode requested_mode;
+
+	if (requested_adv_on) {
+		requested_mode = has_account_key ?
+				 APP_FP_ADV_MODE_NOT_DISCOVERABLE :
+				 APP_FP_ADV_MODE_DISCOVERABLE;
+	} else {
+		requested_mode = APP_FP_ADV_MODE_OFF;
+	}
+
+	/* Switch the advertising mode only on change. */
+	if (requested_mode != fp_adv_mode) {
+		LOG_INF("Fast Pair: advertising: %sabling", requested_adv_on ? "en" : "dis");
+		fp_adv_mode_set(requested_mode);
+	}
 }
 
-static void fp_adv_request_handle(enum app_ui_request request)
+void app_fp_adv_request(struct app_fp_adv_trigger *trigger, bool enable)
 {
-	/* It is assumed that the callback executes in the cooperative
-	 * thread context as it interacts with the FMDN API.
-	 */
-	__ASSERT_NO_MSG(!k_is_preempt_thread());
-	__ASSERT_NO_MSG(!k_is_in_isr());
-
-	if (!app_fp_adv_is_init()) {
+	if (trigger->enabled == enable) {
 		return;
 	}
 
-	if (request == APP_UI_REQUEST_FP_ADV_MODE_CHANGE) {
-		bool has_account_key = bt_fast_pair_has_account_key();
-
-		if (has_account_key) {
-			switch (fp_adv_mode) {
-			case APP_FP_ADV_MODE_OFF:
-				fp_adv_mode = APP_FP_ADV_MODE_NOT_DISCOVERABLE;
-				break;
-			case APP_FP_ADV_MODE_NOT_DISCOVERABLE:
-				fp_adv_mode = APP_FP_ADV_MODE_OFF;
-				break;
-			default:
-				__ASSERT_NO_MSG(0);
-			}
-		} else {
-			switch (fp_adv_mode) {
-			case APP_FP_ADV_MODE_OFF:
-				fp_adv_mode = APP_FP_ADV_MODE_DISCOVERABLE;
-				break;
-			case APP_FP_ADV_MODE_DISCOVERABLE:
-				fp_adv_mode = APP_FP_ADV_MODE_OFF;
-				break;
-			default:
-				__ASSERT_NO_MSG(0);
-			}
-		}
+	trigger->enabled = enable;
 
-		fp_advertising_start();
+	LOG_INF("Fast Pair: advertising request from trigger \"%s\": %s",
+		trigger->id, trigger->enabled ? "enable" : "disable");
+
+	if (is_enabled) {
+		fp_adv_request_start();
+	} else {
+		LOG_WRN("Fast Pair: advertising module is disabled");
 	}
 }
 
+void app_fp_adv_refresh(void)
+{
+	fp_adv_mode_set(fp_adv_mode);
+}
+
+void app_fp_adv_rpa_rotation_suspend(bool suspended)
+{
+	fp_adv_rpa_rotation_suspended = suspended;
+}
+
 int app_fp_adv_id_set(uint8_t id)
```

Comment:
Consider also setting the ID after the app_fp_adv_init(), we should require init to be called before any other function